lib/relation-definition.js

```diff
@@ -1276,8 +1313,12 @@ BelongsTo.prototype.related = function (refresh, params) {
   var scopeQuery = null;
   
   if (arguments.length === 1) {
+    if (typeof refresh === 'boolean') {
+      params = utils.createPromiseCallback();
+    } else {
```

Comment:
This makes `Product.categories(true)` return a promise, correct? I thought we agreed that `Product.categories()` will not be promise-enabled at all. I am proposing to revert all changes made in the `related()` method.